packages/core/admin/admin/src/content-manager/components/BlocksInput/utils/conversions.ts

```diff
@@ -89,4 +91,27 @@ const insertEmptyBlockAtLast = (editor: Editor) => {
   );
 };
 
-export { baseHandleConvert, getAttributesToClear, isLastBlockType, insertEmptyBlockAtLast };
+function useConversionModal() {
+  const [modalComponent, setModalComponent] = React.useState<React.JSX.Element | null>(null);
+
+  const handleConversionResult = (
+    maybeRenderModal: void | (() => React.JSX.Element) | undefined
```

Comment:
aren't these more boolean names? here's it's either the value or undefined. I'm happy to remove the `maybe` though, it's probably more confusing than anything else

Reply:
I guess truthy / falsy names not necessarily boolean. Right that's what your checking here? If I have a component then maybeRenderModal is truthy?

Reply:
I'm actually calling the function, not just checking if it's truthy

The maybe emphasizes that the value may not exist, but I guess Typescript also does that

Reply:
Got it, didn't see the function call I thought it was or was not a value. I guess it's a bit confusing because you may not have a function so you have to check that, then even if you do have a function it may or may not return a modal. Up to you.